apps/babelstream/package.py

```diff
@@ -0,0 +1,363 @@
+    # depends_on("cuda", when="+cuda")
+    #variant("cuda", default=False, description="Enable CUDA support")
+    #variant("cuda_arch", values=str, default="none", description="The CUDA architecture being used.")
+
+
+
+# raise InstallError('This package does not build on Mac OS X!')
+
+    #==> Error: variant "cuda" not found in package "babelstream" [required from package "babelstream"] this line below fixes this
+    #variant("cuda", default=False, description="Enable CUDA support")
+    #variant("cuda_arch", values=str, default="none", description="The CUDA architecture being used.")
+    
+
+
+    variant("sycl", default=False, description="Enable SYCL support")
+    
+
+    #Defined in CudaPackage
+    #variant("cuda", default=False, description="Enable CUDA support")
+    variant("sycl2020", default=False, description="Enable SYCL support")
+    variant("omp", default=False, description="Enable OpenMP support")
+    variant("ocl", default=False, description="Enable OpenCL support") ## Crushes llvm requires RAM
+    variant("tbb", default=False, description="Enable TBB support")
+    variant("acc", default=False, description="Enable OpenACC support") 
+    variant("thrust", default=False, description="Enable THRUST support")
+    variant("raja", default=False, description="Enable RAJA support") 
+
+    variant("stddata", default=False, description="Enable STD-data support")
+    variant("stdindices", default=False, description="Enable STD-data support")
+    variant("stdranges", default=False, description="Enable STD-data support")
+    
+    variant("dir", values=str, default="none", description="Enable Directory support")
+
+    conflicts("dir=none",when="+kokkos",msg="KOKKKOS requires architecture to be specfied by dir=")
+    variant("kokkos", default=False, description="Enable KOKKOS support")
+    
+    #ACC conflict
+    variant("cpu_arch", values=str, default="none", description="Enable CPU Target for ACC")
+    # conflicts("cpu_arch=none",when="+acc",msg="KOKKKOS requires architecture to be specfied by dir=")
+    variant("target", values=str, default="none", description="Enable CPU Target for ACC")
+    
+    
+    # CUDA conflict 
+    # this is not provided in CudaPackage
+    conflicts("cuda_arch=none",when="+cuda",msg="CUDA requires architecture to be specfied by cuda_arch=")
+    variant("mem", values=str, default="DEFAULT", description="Enable MEM Target for CUDA")
+        
+    #download raja from https://github.com/LLNL/RAJA
+    conflicts("dir=none",when="+raja",msg="KOKKKOS requires architecture to be specfied by dir=")
+
+    #Thrust Conflict
+    conflicts("~cuda", when="+thrust", msg="Thrust requires +cuda variant")
+
+    #TBB Dependency
+    depends_on("intel-oneapi-tbb", when="+tbb")
+
+    #depends_on("cmake@3.8", when="+stdindices")
+    depends_on("cmake@3.23.2", when="+stddata")
+    depends_on("gcc@12.1.0", when="+stddata")
```

Comment:
Known issue with `gcc@9.2.0` also shared [here](https://github.com/compiler-explorer/compiler-explorer/issues/2075) it won't work with gcc@9.2.0 gives the following error. So we need  `gcc@12.1.0` on the environment too because adding like a dependency this way won't force it. So priority is always given to environment settings. @giordano 

` >> 90    /cm/local/apps/gcc/9.2.0/include/c++/9.2.0/pstl/parallel_backend_tbb.h:19:10: fatal error: tbb/blocked_range.h: No such file or directory
    19| #include <tbb/blocked_range.h>
       compilation terminated.`

Reply:
If you have problems with compilers, instead of using [`depends_on()`](https://spack.readthedocs.io/en/latest/packaging_guide.html#depends-on) with that compiler (which would make it complicated to build with *other* compilers), you should use [`conflicts()`](https://spack.readthedocs.io/en/latest/packaging_guide.html#conflicts).